File: app/src/main/java/com/djavid/bitcoinrate/model/heroku/Subscribe.java
package com.djavid.bitcoinrate.model.heroku;

import io.realm.RealmObject;


public class Subscribe extends RealmObject {

    private Long id;
    private boolean isTrendingUp;
    private String value;
    private long tickerId;
    private long tokenId;
    private String cryptoId;
    private String countryId;
    private double change_percent;


    public Subscribe() { }

    public Subscribe(boolean isTrendingUp, String value, long tickerId, long tokenId, String cryptoId, String countryId) {

        this.isTrendingUp = isTrendingUp;
        this.value = value;
        this.tickerId = tickerId;
        this.tokenId = tokenId;
        this.cryptoId = cryptoId;
        this.countryId = countryId;

        this.change_percent = 0;
    }

    public Subscribe(String value, long tickerId, long tokenId, String cryptoId, String countryId, double curr_price) {

        this.value = Double.toString(curr_price);
        this.tickerId = tickerId;
        this.tokenId = tokenId;
        this.cryptoId = cryptoId;
        this.countryId = countryId;
        this.change_percent = Double.parseDouble(value);

        this.isTrendingUp = false;
    }

    @Override
    public String toString() {
        return "Subscribe{" +
                "id=" + id +
                ", isTrendingUp=" + isTrendingUp +
                ", value='" + value + '\'' +
                ", tickerId=" + tickerId +
                ", tokenId=" + tokenId +
                ", cryptoId='" + cryptoId + '\'' +
                ", countryId='" + countryId + '\'' +
                ", change_percent=" + change_percent +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;

        Subscribe subscribe = (Subscribe) o;

        if (isTrendingUp != subscribe.isTrendingUp) return false;
        if (tickerId != subscribe.tickerId) return false;
        if (tokenId != subscribe.tokenId) return false;
        if (Double.compare(subscribe.change_percent, change_percent) != 0) return false;
        if (id != null ? !id.equals(subscribe.id) : subscribe.id != null) return false;
        if (value != null ? !value.equals(subscribe.value) : subscribe.value != null) return false;
        if (cryptoId != null ? !cryptoId.equals(subscribe.cryptoId) : subscribe.cryptoId != null)
            return false;
        return countryId != null ? countryId.equals(subscribe.countryId) : subscribe.countryId == null;
    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        result = id != null ? id.hashCode() : 0;
        result = 31 * result + (isTrendingUp ? 1 : 0);
        result = 31 * result + (value != null ? value.hashCode() : 0);
        result = 31 * result + (int) (tickerId ^ (tickerId >>> 32));
        result = 31 * result + (int) (tokenId ^ (tokenId >>> 32));
        result = 31 * result + (cryptoId != null ? cryptoId.hashCode() : 0);
        result = 31 * result + (countryId != null ? countryId.hashCode() : 0);
        temp = Double.doubleToLongBits(change_percent);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        return result;
    }

    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }

    public boolean isTrendingUp() {
        return isTrendingUp;
    }
    public void setTrendingUp(boolean trendingUp) {
        this.isTrendingUp = trendingUp;
    }

    public String getValue() {
        return value;
    }
    public void setValue(String value) {
        this.value = value;
    }

    public long getTickerId() {
        return tickerId;
    }
    public void setTickerId(long tickerId) {
        this.tickerId = tickerId;
    }

    public String getCryptoId() {
        return cryptoId;
    }
    public void setCryptoId(String cryptoId) {
        this.cryptoId = cryptoId;
    }

    public String getCountryId() {
        return countryId;
    }
    public void setCountryId(String countryId) {
        this.countryId = countryId;
    }

    public long getTokenId() {
        return tokenId;
    }
    public void setTokenId(long tokenId) {
        this.tokenId = tokenId;
    }

    public double getChange_percent() {
        return change_percent;
    }
    public void setChange_percent(double change_percent) {
        this.change_percent = change_percent;
    }
}

File: app/src/main/java/com/djavid/bitcoinrate/model/heroku/Subscribe.kt
package com.djavid.bitcoinrate.model.heroku

import io.realm.RealmObject


open class Subscribe : RealmObject {

    var id: Long? = null
    var isTrendingUp: Boolean = false
    var value: String? = null
    var tickerId: Long = 0
    var tokenId: Long = 0
    var cryptoId: String? = null
    var countryId: String? = null
    var change_percent: Double = 0.toDouble()


    constructor()

    constructor(isTrendingUp: Boolean, value: String, tickerId: Long, tokenId: Long, cryptoId: String, countryId: String) {

        this.isTrendingUp = isTrendingUp
        this.value = value
        this.tickerId = tickerId
        this.tokenId = tokenId
        this.cryptoId = cryptoId
        this.countryId = countryId

        this.change_percent = 0.0
    }

    constructor(value: String, tickerId: Long, tokenId: Long, cryptoId: String, countryId: String, curr_price: Double) {

        this.value = java.lang.Double.toString(curr_price)
        this.tickerId = tickerId
        this.tokenId = tokenId
        this.cryptoId = cryptoId
        this.countryId = countryId
        this.change_percent = java.lang.Double.parseDouble(value)

        this.isTrendingUp = false
    }

    override fun toString(): String {
        return "Subscribe{" +
                "id=" + id +
                ", isTrendingUp=" + isTrendingUp +
                ", value='" + value + '\''.toString() +
                ", tickerId=" + tickerId +
                ", tokenId=" + tokenId +
                ", cryptoId='" + cryptoId + '\''.toString() +
                ", countryId='" + countryId + '\''.toString() +
                ", change_percent=" + change_percent +
                '}'.toString()
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null) return false

        val subscribe = o as Subscribe?

        if (isTrendingUp != subscribe!!.isTrendingUp) return false
        if (tickerId != subscribe.tickerId) return false
        if (tokenId != subscribe.tokenId) return false
        if (java.lang.Double.compare(subscribe.change_percent, change_percent) != 0) return false
        if (if (id != null) id != subscribe.id else subscribe.id != null) return false
        if (if (value != null) value != subscribe.value else subscribe.value != null) return false
        if (if (cryptoId != null) cryptoId != subscribe.cryptoId else subscribe.cryptoId != null)
            return false
        return if (countryId != null) countryId == subscribe.countryId else subscribe.countryId == null
    }

    override fun hashCode(): Int {
        var result: Int
        val temp: Long
        result = if (id != null) id!!.hashCode() else 0
        result = 31 * result + if (isTrendingUp) 1 else 0
        result = 31 * result + if (value != null) value!!.hashCode() else 0
        result = 31 * result + (tickerId xor tickerId.ushr(32)).toInt()
        result = 31 * result + (tokenId xor tokenId.ushr(32)).toInt()
        result = 31 * result + if (cryptoId != null) cryptoId!!.hashCode() else 0
        result = 31 * result + if (countryId != null) countryId!!.hashCode() else 0
        temp = java.lang.Double.doubleToLongBits(change_percent)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        return result
    }
}
